### ISSUE TITLE
setEmail and logoutEmail cause entire app to freeze

### ISSUE THREAD

**Issue by burdell** [OneSignal/react-native-onesignal#601]
**Description:**
Calling `setEmail` or `logoutEmail` causes the entire app to freeze in Android. Not seeing the same thing happening for iOS

**Environment**
1. react-native: 0.55.4
2. react-native-onesignal: 3.2.5
3. Tested on real Samsung Galaxy s9 and emulated Pixel 2.7
4. App is in beginning stages so don't have a real signed staging app to test against (not sure if that matters)

**Steps to Reproduce Issue:**
1. Login to our app - flow calls `setEmail` and app totally freezes
2. Restart the app and see it load data verifying the login function did work
3. Logout - flow calls `logoutEmail` and app totally freezes again
4. Restart the app and see logout was successful

**Anything else:**
- Doing `abd logcat` while running the emulator shows that there is a recurring log message stating  `NetRecChimeraGcmTaskService.a: Still entries pending update, scheduling new rapid_refresh_scores_task refresh.` which kinda makes think that the call never completes and just hangs the app up 🤷‍♂️

**Comment by Nightsd01** [comment]
@burdell Are you getting an ANR? Can you post it please? Unfortunately I am not able to reproduce this problem.

**Comment by burdell** [comment]
@Nightsd01 

after doing a little more digging, I think the freezing part was a weird condition of logging in and out of various devices with a few different testing logins...couldn't actually figure out the root cause of that was

After I went through and just cleared out all the test logins / devices from our OneSignal staging feed, I DID find that logging into iOS adds the device to the push notification feed, but logging into the Android app adds the device to the email notification feed. So that makes sense that I can get iOS notifications, but not Android.

We're doing push notifications in our current Ionic app by the associated user's email (not sure if that's standard practice or the reasoning behind that...I inherited this notification setup a few weeks ago). Trying to just hook up our re-written React Native app to the current OneSignal feeds

Right now I'm calling both `sendTag` and `setEmail`:
```
OneSignal.sendTag("email", userEmail);
OneSignal.setEmail(userEmail, () => ({}));
```
Not sure if this is the correct way to do this...most documentation seems to lean towards using the corresponding device id.

Any direction here? Maybe I close this and open a new issue?
